Components/Queues/Serialization/Serializer.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Smartbox\Integration\FrameworkBundle\Components\Queues\Serialization;
+
+use JMS\Serializer\Exception\RuntimeException;
+use JMS\Serializer\SerializerInterface;
+use Smartbox\CoreBundle\Type\SerializableInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Messages\MessageInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Serializers\QueueSerializerInterface;
+use Smartbox\Integration\FrameworkBundle\Exceptions\MessageDecodingFailedException;
+
+class Serializer implements QueueSerializerInterface
+{
+    const FORMAT_JSON = 'json';
+
+    private $serializer;
+    private $format;
+
+    public function __construct(SerializerInterface $serializer, string $format = self::FORMAT_JSON)
+    {
+        $this->serializer = $serializer;
+        $this->format = $format;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function decode(array $encodedMessage): MessageInterface
+    {
+        if (empty($encodedMessage['body']) || !is_array($encodedMessage['headers'])) {
+            throw new MessageDecodingFailedException('Encoded message should have at least a "body" and some "headers"');
+        }
+
+        try {
+            $message = $this->serializer->deserialize($encodedMessage['body'], SerializableInterface::class, $this->format);
+        } catch (RuntimeException $e) {
```

Comment:
Would it make sense to catch the `InvalidArgumentException` from doctrine here as well? Currently this one just rethrows another one, and on EAI's side we check for doctrine's as well... Since this is thrown by JSM, it might be wise to catch it here as well.

Reply:
This would simplify EAI's side as well, as it only needs to check for one exception only.

Reply:
Added